### PR TITLE
WINDUP-493: Add annotation scanning support to EJB discovery

### PR DESCRIPTION
**Pull request by jsight**
https://issues.jboss.org/browse/WINDUP-493

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/468/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.tests:tests' /><a href='https://10.3.9.192/job/windup-pr-builder/468/org.jboss.windup.tests$tests/testReport'>org.jboss.windup.tests:tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://10.3.9.192/job/windup-pr-builder/468/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureMediumBinaryModeTest/testRunWindupMedium/'><strong>org.jboss.windup.tests.application.WindupArchitectureMediumBinaryModeTest.testRunWindupMedium</strong></a></li></ul>
Test FAILed.

File: rules-java-ee/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEjbAnnotationsRuleProvider.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                    .addRule()
+                    .when(JavaClass.references("javax.ejb.{annotationType}").at(TypeReferenceLocation.ANNOTATION))
+                    .perform(new AbstractIterationOperation<JavaTypeReferenceModel>()
+                    {
+                        public void perform(GraphRewrite event, EvaluationContext context, JavaTypeReferenceModel payload)
+                        {
+                            extractEJBMetadata(event, payload);
+                        };
+                    })
+                    .where("annotationType").matches("Stateless|Stateful")
```
Comment (lincolnthree) [inline rules-java-ee/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEjbAnnotationsRuleProvider.java]: Very nice!

Comment (jsight) [inline rules-java-ee/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEjbAnnotationsRuleProvider.java]: Yeah, this parameter stuff is really useful.

**Comment by lincolnthree** [comment]
Looks good. We can fix the OR issue later. As you said, it's tested, so we can rely on that if the short-circuiting functionality ever goes in.

**Comment by jsight** [comment]
Yeah, I think all we need is a reliable way to register TypeInterest in getConfiguration. At the moment, anything registered in there will get wiped out.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/479/
Test PASSed.

**Pull request merged**